modules/upsell-order-bump/includes/OrderBump.php

```
@@ -121,4 +159,122 @@ public function woocommerce_custom_price_to_cart_item( $cart_object ) {
+
+		foreach ( $cart->get_cart() as $cart_item_key => $cart_item ) {
+			$cart_product_id = $cart_item['product_id'];
+			$cart_variation_id = isset( $cart_item['variation_id'] ) ? $cart_item['variation_id'] : 0;
+			$cart_item_id = $cart_variation_id > 0 ? $cart_variation_id : $cart_product_id;
+
+			// Check if this is the bump product with custom pricing and remove it
+			if ( $cart_item_id == $offer_product_id && isset( $cart_item['custom_price'] ) ) {
+				$cart->remove_cart_item( $cart_item_key );
+			}
+		}
+	}
```

PHPMD (2.15.0)

Avoid unused private methods such as 'handle_orphaned_bump_product'. (Unused Code Rules)

(UnusedPrivateMethod)